cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -0,0 +1,363 @@
+    is applied between each and every pair of qubit that are next to each
+    other. In general, a total of four sets of parallel operations are needed to
+    perform all possible two-qubit gates. We proceed as follows:
+
+    The first layer applies two-qubit gates to qubits (i, j) and (i, j + 1)
+    where i is any integer and j is an even integer. The second layer
+    applies two-qubit gates to qubits (i, j) and (i + 1, j) where i is an even
+    integer and j is any integer. The third layer applies two-qubit gates
+    to qubits (i, j) and (i, j + 1) where i is any integer and j is an odd
+    integer. The fourth layer applies two-qubit gates to qubits (i, j) and
+    (i + 1, j) where i is an odd integer and j is any integer.
+
+    After the layers are built as above, any empty layer is ejected.:
+
+                 Cycle 1:                            Cycle 2:
+        q00 ── q01    q02 ── q03            q00    q01    q02    q03
+                                             |      |      |      |
+        q10 ── q11    q12 ── q13            q10    q11    q12    q13
+
+        q20 ── q21    q22 ── q23            q20    q21    q22    q23
+                                             |      |      |      |
+        q30 ── q31    q32 ── q33            q30    q31    q32    q33
+
+                  Cycle 3:                           Cycle 4:
+        q00    q01 ── q02    q03            q00    q01    q02    q03
+
+        q10    q11 ── q12    q13            q10    q11    q12    q13
+                                             |      |      |      |
+        q20    q21 ── q22    q23            q20    q21    q22    q23
+
+        q30    q31 ── q32    q33            q30    q31    q32    q33
+
+    Args:
+        qubits: The grid qubits included in the entangling operations.
+        two_qubit_gate: The two-qubit gate to be applied between all
+            neighboring pairs of qubits.
+
+    Returns:
+        A list of sets of operations, with a maximum length of 4. Each set
+        includes two-qubit gates which can be performed at the same time.
+    """
+    interaction_sequence = _default_interaction_sequence(qubits)
+    return [{two_qubit_gate(q_a, q_b) for (q_a, q_b) in pairs} for pairs in
+            interaction_sequence]
+
+
+def _build_xeb_circuits(qubits: Sequence[ops.Qid],
+                        cycles: Sequence[int],
+                        single_qubit_gates: List[List[
+                            ops.SingleQubitGate]] = None,
+                        benchmark_ops: Sequence[Set[ops.OP_TREE]] = None,
+                        ) -> List[circuits.Circuit]:
+    if benchmark_ops is not None:
+        num_d = len(benchmark_ops)
+    else:
+        num_d = 0
+    max_cycles = max(cycles)
+
+    if single_qubit_gates is None:
+        single_rots = _random_half_rotations(qubits, max_cycles)
+    else:
+        single_rots = _random_any_gates(qubits, single_qubit_gates, max_cycles)
+    all_circuits = []  # type: List[circuits.Circuit]
```

Comment:
As of recently can write `all_circuits: List[circuits.Circuits] = []`.

Reply:
I have changed the XEB function to the linear version, which circumvents the issue of artificially clipping the probabilities to some small value. The code and doc-strings have all been modified to reflect this.